pinc/misc.inc

```diff
@@ -1272,6 +1272,38 @@ function factor_strings($strings)
     return [$left_common, $middles, $right_common];
 }
 
+//--------------------------------------------------------------------------
+
+/**
+ * Given a function and arguments that will generate data, attempt
+ * to load the data from memcached first and fall back to the function and
+ * cache the response. If memcache is not running, this will call the requested
+ * function and return its data without generating a notice/warning/error.
+ *
+ * If the requested function returns data that has localized strings, it's
+ * important that $key_salt = get_desired_language();
+ */
+function memoize_function($function, $args = [], $expire_from_now = 3600, $key_salt = "")
+{
+    $memcache = new Memcached();
+    $memcache->addServer('localhost', 11211);
```

Comment:
what if there is no memcache server? Is this setup in configuration somewhere?

Reply:
If there is no local memcache server running, or it's not connected, all of the memcache calls return `false` and the code falls through to just calling the function, I tried to convey this in the function header with:
```
If memcache is not running, this will call the requested
function and return its data without generating a notice/warning/error.
```

So having memcache is effectively optional and if it isn't there or isn't working there won't be any caching.

The API uses memcache for rate limiting and this is documented in the `API.md`. I should update `INSTALL.md` and mention the use of memcache for other caching too, I'll do that.

Reply:
I've updated both `API.md` and `INSTALL.md` to include much more details about memcached usage.

Reply:
Perfect, thank you!